### ISSUE TITLE
Cargo Subcommand?

### ISSUE THREAD

**Issue by c-git** [emilk/eframe_template#108]
# TLDR

Would it be desirable to have a [cargo subcommand](https://doc.rust-lang.org/book/ch14-05-extending-cargo.html) to create a new project instead of only this template?

# Explanation

I'm about to start a new egui project and was about to copy the template again. Then manually make the required changes again and realized this is probably something that should get automated. I'm considering building a cargo subcommand that would:

1. Ask the user for a project name and run cargo init to create the new project
2. Download the files from the master branch of this template
3. Prompt the user for answers and make the changes specified in the [getting started](https://github.com/emilk/eframe_template#getting-started)
4. Prompt the user if caching should be disabled
5. Prompt if "presentation only" files should be deleted, like the readme. (Actually this is the only one I can think of right now but I'd go through the files and check)

This would provide a quick way to spin up a new project and allow users and easy way to opt out of caching at it appears [some ppl would prefer not use it](https://github.com/emilk/eframe_template/issues/84#issuecomment-1413086281). It would also make it easier for ppl who are only testing and do not want caching at all like in this [issue](https://github.com/emilk/eframe_template/issues/99).

**Comment by c-git** [comment]
After [further reading}(https://github.com/emilk/eframe_template/pull/102#issuecomment-1682598515) I'm wondering if we shouldn't default to no caching as it seems to causes issue for a lot of ppl (tbh I actually like it, but I've not deployed any of my other apps yet so I don't know how it works in production).  And using the wizard from the subcommand ppl can opt in.

**Comment by c-git** [comment]
Abandoning this.
